sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala

```diff
@@ -19,16 +19,33 @@ package org.apache.spark.sql.execution.streaming
 
 import org.apache.spark.sql.catalyst.expressions.Attribute
 import org.apache.spark.sql.catalyst.plans.logical.LeafNode
+import org.apache.spark.sql.execution.datasources.DataSource
 
 object StreamingRelation {
-  def apply(source: Source): StreamingRelation =
-    StreamingRelation(source, source.schema.toAttributes)
+  def apply(dataSource: DataSource): StreamingRelation = {
+    val source = dataSource.createSource()
+    StreamingRelation(dataSource, source.schema.toAttributes)
+  }
+}
+
+/**
+ * Used to link a streaming [[DataSource]] into a
+ * [[org.apache.spark.sql.catalyst.plans.logical.LogicalPlan]].
+ */
+case class StreamingRelation(dataSource: DataSource, output: Seq[Attribute]) extends LeafNode {
+  override def toString: String = dataSource.createSource().toString
```

Comment:
This could be expensive right?  I'm not sure that we want to do that in a `toString`.